library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -375,8 +378,30 @@ private boolean updateLocked() {
         return isActive;
     }
 
+    private boolean isBatchStartingForTheFirstTime(DownloadBatch downloadBatch) {
+        boolean isBatchWithoutWrittenBytes = downloadBatch.getCurrentSize() == 0;
+
+        long batchId = downloadBatch.getBatchId();
+        if (batchWithoutWrittenBytesAlreadyChecked(isBatchWithoutWrittenBytes, batchId)) {
```

Comment:
Maybe I am missing something but I'm not sure why you need this, could you explain this for me?

To me, it looks like you should return in the statements below.